check_test.go

```diff
@@ -258,6 +278,73 @@ func getCheckTestCases() map[string]TmpCheckTestCase {
 				input.AlertSourceNamePredicate = predicateUpdateInput
 				return c.UpdateCheckAlertSourceUsage(*input)
 			},
+			expectedCheck: CheckWithExtras(map[string]any{
+				"alertSourceNamePredicate": predicateUpdateInput,
+				"alertSourceType":          ol.AlertSourceTypeEnumDatadog,
+			}),
+		},
+
+		"CreateCodeIssue": {
+			fixture: BuildCreateRequest("CodeIssue", map[string]any{
+				"constraint":     "exact",
+				"issueName":      "test-issue",
+				"issueType":      []string{"bug", "error"},
+				"maxAllowed":     3,
+				"resolutionTime": map[string]any{"unit": "day", "value": 1},
+				"severity":       []string{"sev1", "sev2"},
+			}),
+			body: func(c *ol.Client) (*ol.Check, error) {
+				input := ol.NewCheckCreateInputTypeOf[ol.CheckCodeIssueCreateInput](checkCreateInput)
+				input.Constraint = ol.CheckCodeIssueConstraintEnumExact
+				input.IssueName = ol.RefOf("test-issue")
+				input.IssueType = ol.RefOf([]string{"bug", "error"})
+				input.MaxAllowed = ol.RefOf(3)
+				input.ResolutionTime = ol.RefOf(ol.CodeIssueResolutionTimeInput{
+					Unit:  ol.CodeIssueResolutionTimeUnitEnumDay,
+					Value: 1,
+				})
+				input.Severity = ol.RefOf([]string{"sev1", "sev2"})
+				return c.CreateCheckCodeIssue(*input)
+			},
+			expectedCheck: CheckWithExtras(map[string]any{
+				"constraint":     "exact",
+				"issueName":      "test-issue",
+				"issueType":      []string{"bug", "error"},
+				"maxAllowed":     3,
+				"resolutionTime": map[string]any{"unit": "day", "value": 1},
+				"severity":       []string{"sev1", "sev2"},
+			}),
+		},
+		"UpdateCodeIssue": {
```

Comment:
I think we'll want to throw a few more "update" test cases in here to ensure it generates the right input for a few of these optional fields.  `maxAllowed`, `resolutionTime` etc etc